src/app/filelogger.cpp

```diff
@@ -0,0 +1,267 @@
+}
+
+int FileLogger::ageType()
+{
+    int val = settings()->loadValue(KEY_AGETYPE, 1).toInt();
+    return (val < 0 || val > 2) ? 1 : val;
+}
+
+void FileLogger::setAgeType(const int &value)
+{
+    settings()->storeValue(KEY_AGETYPE, (value < 0 || value > 2) ? 1 : value);
+}
+
+void FileLogger::configure()
+{
+    QString tmpPath = Utils::Fs::fromNativePath(path());
+    if (!tmpPath.endsWith('/'))
+        tmpPath += "/";
+    QDir dir(tmpPath);
+    dir.mkpath(tmpPath);
+    tmpPath += "qbittorrent.log";
+
+    if (tmpPath != m_path) {
+        m_path = tmpPath;
+
+        if (m_logFile) {
+            closeLogFile();
+            delete m_logFile;
+        }
+        m_logFile = new QFile(m_path);
+        openLogFile();
+    }
+
+    if (isDeleteOld()) {
+        QDateTime date = QDateTime::currentDateTime();
+
+        switch (static_cast<FileLogAgeType>(ageType())) {
+        case DAYS:
+            date = date.addDays(age());
+            break;
+        case MONTHS:
+            date = date.addMonths(age());
+            break;
+        default:
+            date = date.addYears(age());
+        }
+
+        foreach (const QFileInfo file, dir.entryInfoList(QStringList() << "qbittorrent.log.bak*", QDir::Files | QDir::Writable, QDir::Time | QDir::Reversed)) {
+            if (file.lastModified() < date)
+                break;
+            Utils::Fs::forceRemove(file.absoluteFilePath());
+        }
+    }
+}
+
+void FileLogger::addLogMessage(const Log::Msg &msg)
+{
+    if (!m_logFile) return;
+
+    QTextStream str(m_logFile);
+    QString type;
+
+    switch (msg.type) {
+    case Log::INFO:
+        type = "(I) ";
+        break;
+    case Log::WARNING:
+        type = "(W) ";
+        break;
+    case Log::CRITICAL:
+        type = "(C) ";
+        break;
+    default:
+        type = "(N) ";
+    }
+
+    str << type << QDateTime::fromMSecsSinceEpoch(msg.timestamp).toString(Qt::ISODate) << " - " << msg.message << endl;
+
+    if (isBackup() && (m_logFile->size() >= (maxSize() * 1024 * 1024))) {
+        closeLogFile();
+        int counter = 0;
+        QString backupLogFilename = m_path + ".bak";
+
+        while (QFile::exists(backupLogFilename)) {
+            ++counter;
+            backupLogFilename = m_path + ".bak" + QString::number(counter);
+        }
+
+        QFile::rename(m_path, backupLogFilename);
+        openLogFile();
+    }
+    else {
+        m_flusher.start();
+    }
+}
+
+void FileLogger::flushLog()
+{
+    if (m_logFile)
+        m_logFile->flush();
+}
+
+void FileLogger::openLogFile()
+{
+    if (!m_logFile->open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text)
+        || !m_logFile->setPermissions(QFile::ReadOwner | QFile::WriteOwner)) {
+        delete m_logFile;
+        m_logFile = nullptr;
+        Logger::instance()->addMessage(tr("An error occured while trying to open the filelog. Logging to file is disabled."), Log::CRITICAL);
```

Comment:
@sledgehammer999:  Try `log-file` or `log file` instead.  It is more gramatically correct.